KamerConnect.DataAccess.Postgres/Repositorys/PersonRepository.cs

```diff
@@ -202,4 +207,121 @@ private static T ValidateEnum<T>(string value) where T : struct
 
         throw new ArgumentException($"Invalid value '{value}' for enum {typeof(T).Name}");
     }
+
+    public void UpdatePerson(Person person)
+    {
+        using (var connection = new NpgsqlConnection(ConnectionString))
+        {
+            connection.Open();
+            using (var command = new NpgsqlCommand(
+                       """
+                       UPDATE person SET
+                           email = @Email,
+                           first_name = @FirstName,
+                           middle_name = @MiddleName,
+                           surname = @Surname,
+                           phone_number = @PhoneNumber,
+                           birth_date = @BirthDate,
+                           gender = @gender::gender,
+                           role = @Role::user_role,
+                           profile_picture_path = @ProfilePicturePath
+                       WHERE id = @Id;
+                       """, connection))
+            {
+                command.Parameters.AddWithValue("@Id", Guid.Parse(person.Id));
+                command.Parameters.AddWithValue("@Email", person.Email);
+                command.Parameters.AddWithValue("@FirstName", person.FirstName);
+                command.Parameters.AddWithValue("@MiddleName", person.MiddleName ?? string.Empty);
+                command.Parameters.AddWithValue("@Surname", person.Surname);
+                command.Parameters.AddWithValue("@PhoneNumber", person.PhoneNumber ?? string.Empty);
+                command.Parameters.AddWithValue("@BirthDate", person.BirthDate);
+                command.Parameters.AddWithValue("@Gender", person.Gender.ToString());
+                command.Parameters.AddWithValue("@Role", person.Role.ToString());
+                command.Parameters.AddWithValue("@ProfilePicturePath", person.ProfilePicturePath ?? string.Empty);
+
+                command.ExecuteNonQuery();
+            }
+        }
+    }
+    
+    public void UpdatePersonality(Person person)
```

Comment:
Update personlity kan je mogelijk aanroepen in person, dan hoef je die niet nog een keer los aanteroepen

Reply:
Snap ik, is helemaal goed. Mischien alsnog aanroepen in de Person en de andere los houden. Als je dan een UpdatePerson object aanroept werkt die alles wat in het object bij. Dit kan mischien toekomstige bugs voorkomen in de toekomst met betrekking tot personen update.